db/view/view.cc

```diff
@@ -342,7 +342,7 @@ class view_updates final {
             , _updates(8, partition_key::hashing(*_view), partition_key::equality(*_view)) {
     }
 
-    void move_to(std::vector<frozen_mutation_and_schema>& mutations) && {
+    void move_to(utils::chunked_vector<frozen_mutation_and_schema>& mutations) && {
```

Comment:
"The number of view updates can grow large". We won't have a large allocation now with chunked_vector, but what prevents OOM or just extreme memory stress?

Reply:
Nothing at the moment, that's why none of today's patches actually close the issue. Well, maybe this one slightly prevents bad allocs because the allocations are at least in smaller chunks.

Reply:
Ok.

Reply:
I'm rusty with this code, so maybe I'm asking for a huge rewrite so feel free to tell me know, but why do we need to prepare all these mutations in a vector (chunked or otherwise) up-front? Why can't we just prepare up to (say) 128 of them and then update each of them, using a semaphore or whatever as proposed in the next patch?

Reply:
It's a small rewrite indeed, my initial idea was to create easily backportable patches that soothe the problem and then think of a way of actually fixing issue #8852. But perhaps it won't be that intrusive, now that I have a fresh morning look over the code. Specifically, this code:
https://github.com/scylladb/scylla/blob/9d49a26e79e675a1296c30dc55cbf023ed6c872f/db/view/view.cc#L901-L923
already has a big `repeat` loop inside it, which would be a natural candidate for returning partial results from the function instead of accumulating all of them in a vector. I think that this series makes sense on its own too, but on top of it I'll try to rework the way view updates are generated so that we don't need to fill the whole vector (chunked or not) in the first place.